src/modules/sessionManager.js

```diff
@@ -71,6 +70,19 @@ class SessionManager {
      * @returns A promise from the window.fetch API
      */
     gatekeeper = (path, request) => SessionManager.#fetch(this.gatekeeperUrl, path, request);
+
+    /**
+     * Convert a fetch response to json when the result was valid.
+     *
+     * @param {*} response
+     * @returns json as object or a rejected Promise when the response status was not 2xx or the Content-Type was not application/json
+     */
+    toJson = (response) => {
+        if (response.status >= 200 && response.status < 300 && response.headers.get('Content-Type').indexOf('application/json') === 0) {
+            return response.json();
```

Comment:
It's kind of an inverse guard now. I would expect this to be the exceptional case and the return at the end to be the happy flow.

Reply:
ok